fix: #1126 More appealing empty history screen

bhattabhi013 opened the pull request:
### What
More appealing empty history screen

### Screenshot
<img width="233" alt="image" src="https://user-images.githubusercontent.com/47862474/158325378-ef27c38a-3820-4c61-9fb6-fa28ec2d4728.png">

### Fixes bug(s)
- #1126 
### Part of 
- https://github.com/openfoodfacts/smooth-app/issues/525
(please be as granular as possible)

bhattabhi013 committed:
fixed #1126 More appealing empty history screen

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -122,10 +122,42 @@ class _ProductListPageState extends State<ProductListPage> {
+                )
+              ],
             )
+          // ? Center(
+          //     child: Text(appLocalizations.no_prodcut_in_list,
+          //         style: Theme.of(context).textTheme.subtitle1),
+          //   )
```
Please remove those comments.

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -122,10 +122,42 @@ class _ProductListPageState extends State<ProductListPage> {
+                Text(
+                  'Start scanning!',
+                  style: TextStyle(
+                    color: Colors.black,
```
Careful with colors: we also have a dark mode.

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -122,10 +122,42 @@ class _ProductListPageState extends State<ProductListPage> {
+                  }),
+                ),
+                Text(
+                  'Start scanning!',
```
Please add a `TODO` regarding localization.

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -122,10 +122,42 @@ class _ProductListPageState extends State<ProductListPage> {
+                  style: TextStyle(
+                    color: Colors.black,
+                    fontWeight: FontWeight.bold,
+                    fontSize: 36.0,
```
Please don't use random values for fonts: use theme values instead.

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -122,10 +122,42 @@ class _ProductListPageState extends State<ProductListPage> {
+                  ),
+                ),
+                Padding(
+                  padding: EdgeInsets.all(20.0),
```
Use `VERY_LARGE_SPACE` from `design_constants.dart` instead.

monsieurtanuki reviewed (commented):
Thank you @bhattabhi013 for this PR!
Please read my comments.
In addition to them:
* I think the whole block should be vertically centered, which was not obvious in your code or in your screenshot
* regarding the PR title, please use "fix: " instead, as this will be used for automatic PR sorting
* I'm quite sure that your PR doesn't look good in dark mode; please add screenshots in both day and night mode

bhattabhi013 commented:
Thanks for the feedback @monsieurtanuki . I'll resolve the pending issues so that the PR can be merged.

bhattabhi013 committed:
fix #1126

bhattabhi013 committed:
Merge branch 'develop' into develop

bhattabhi013 commented:
I've fixed the pending issues and updated the changes. 
Please find the screenshots:
1) day mode
<img width="241" alt="image" src="https://user-images.githubusercontent.com/47862474/158361206-fcc7c2ce-88e5-4021-884f-59e4610c9460.png">
2) night mode
<img width="244" alt="image" src="https://user-images.githubusercontent.com/47862474/158361300-bd06901b-27b9-4c4e-9f46-a8f3b5994fbb.png">

M123-dev commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -13,6 +13,8 @@ import 'package:smooth_app/pages/personalized_ranking_page.dart';
 import 'package:smooth_app/pages/product/common/product_list_item_simple.dart';
 import 'package:smooth_app/pages/product/common/product_query_page_helper.dart';
 
+import '../../../generic_lib/design_constants.dart';
```
Please use a direct not a relative import here

M123-dev commented on packages/smooth_app/lib/pages/product/common/product_list_page.dart:
```diff
@@ -123,9 +126,33 @@ class _ProductListPageState extends State<ProductListPage> {
+                  child: Icon(
+                    Icons.find_in_page_rounded,
+                    color: colorScheme.primary,
+                    size: VERY_LARGE_SPACE * 10,
```
This could get out of bounds of the SizedBox, since VERY_LARGE_SPACE  isn't relative to the devices size

bhattabhi013 replied:
What should I use instead? I was thinking to use container.

monsieurtanuki replied:
@bhattabhi013 What about screen width / 2?

M123-dev replied:
Then we probably don't need the SizedBox anymore

bhattabhi013 replied:
I've removed sized box as suggested by @M123-dev .

M123-dev reviewed (commented):
All in all looks good, just two small comments

bhattabhi013 committed:
fix #1126

bhattabhi013 committed:
Merge branch 'develop' of https://github.com/bhattabhi013/smooth-app into develop

bhattabhi013 committed:
Merge branch 'develop' into develop

M123-dev reviewed (approved):
Thanks @bhattabhi013, looks good
